pylsp_mypy/plugin.py

```diff
@@ -165,6 +165,20 @@ def match_exclude_patterns(document_path: str, exclude_patterns: list) -> bool:
     return False
 
 
+def get_cmd(settings: Dict[str, Any], cmd: str) -> List[str]:
+    """Get the command to run from settings, falling back to searching the PATH."""
+    command_key = f"{cmd}_command"
+    command: List[str] = settings.get(command_key, [])
+
+    if not command and shutil.which(cmd):
```

Comment:
This would be easier to comprehend:

```python
    if command:
        return command

    if shutil.which(cmd):
        ...
        return [cmd]

    return command
```

Also, it looks like a hidden bug: if `command` is empty and `shutil.which(cmd) == False`, empty list would be returned, which is undesirable

Reply:
> This would be easier to comprehend:
> 
> ```python
>     if command:
>         return command
> 
>     if shutil.which(cmd):
>         ...
>         return [cmd]
> 
>     return command
> ```

I don't think it is actually that much clearer. I think the version with fewer branches is simpler and easier to read.

> Also, it looks like there is a hidden bug: if `command` is empty and `shutil.which(cmd) == False`, empty list would be returned, which is undesirable

That's not a bug. It is intentional and consistent with current behaviour. On lines [323](https://github.com/python-lsp/pylsp-mypy/pull/91/files#diff-022b33447bd6c6eac62326d67e06dea5fe0d42cc11205bafd164ffc8a85cd096R323), [347](https://github.com/python-lsp/pylsp-mypy/pull/91/files#diff-022b33447bd6c6eac62326d67e06dea5fe0d42cc11205bafd164ffc8a85cd096R347) and [386](https://github.com/python-lsp/pylsp-mypy/pull/91/files#diff-022b33447bd6c6eac62326d67e06dea5fe0d42cc11205bafd164ffc8a85cd096R386) we check whether the returned command is truthy (i.e. the command is configured in the settings or it exists on the PATH). If not, we fall back to using the API from the version of mypy installed in the pylsp-mypy environment (see lines [333](https://github.com/python-lsp/pylsp-mypy/pull/91/files#diff-022b33447bd6c6eac62326d67e06dea5fe0d42cc11205bafd164ffc8a85cd096R333), [370](https://github.com/python-lsp/pylsp-mypy/pull/91/files#diff-022b33447bd6c6eac62326d67e06dea5fe0d42cc11205bafd164ffc8a85cd096R370) and [396](https://github.com/python-lsp/pylsp-mypy/pull/91/files#diff-022b33447bd6c6eac62326d67e06dea5fe0d42cc11205bafd164ffc8a85cd096R396)).

I'll update the docstring to make it clearer that this is intentional behaviour.